### PR TITLE
329 create lobby menu

### PR DESCRIPTION
**Pull request by osvegn**
## New features
- New menu to create a room
- Show available rooms and connect to them
## Updated features
- Add text component in button entity

### CODE REVIEW

File: Client/ClientRoom.cpp [hunk 1]
```diff
@@ -150,12 +146,8 @@ void ClientRoom::_startConnexionProtocol(void)
     _communicatorInstance.get()->sendDataToAClient(_serverEndpoint, networkData, sizeof(char) * 10, 14);
     std::free(networkData);
 }
```
Comment (31Nathan) [inline Client/ClientRoom.cpp:148]: Missing space I think

File: Client/ClientRoom.cpp [hunk 2]
```diff
@@ -360,6 +322,8 @@ void ClientRoom::_updateEcsResources(bool isSolo)
         _worldInstance->addResource<GraphicsTextureResource>();
     if (!_worldInstance->containsResource<ButtonActionMap>())
         _worldInstance->addResource<ButtonActionMap>();
+    if (!_worldInstance->containsResource<GameLevel>())
```
Comment (31Nathan) [inline Client/ClientRoom.cpp:325]: Already two lines after

File: libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateButton.hpp [hunk 3]
```diff
@@ -25,9 +25,11 @@ namespace ecs
     /// @param actionName The action name to be set for the Button
     /// @param layerLvl The button layer level, useful for graphical part
     /// @param displayState The associated state to decide when the button will be displayed
+    /// @param text The text to be set on a button.
```
Comment (LucasTesnier) [inline libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateButton.hpp:28]: Tabulation problem?

File: libs/R-TypeLogic/EntityManipulation/ButtonManipulation/SharedResources/ButtonActionMap.hpp [hunk 4]
```diff
@@ -19,7 +19,7 @@ namespace ecs
     class ButtonActionMap : public Resource {
       public:
         /// @brief Enumeration of all the available actions
-        enum buttonAction_e { UNDEFINED, EXIT, PAUSE, RESUME, WRITABLE, WRITABLE_BUTTON, LOBBY };
+        enum buttonAction_e { UNDEFINED, EXIT, PAUSE, RESUME, WRITABLE, WRITABLE_BUTTON, LOBBY, ROOM_CONNECTION, CREATE_A_ROOM, IN_GAME};
```
Comment (LucasTesnier) [inline libs/R-TypeLogic/EntityManipulation/ButtonManipulation/SharedResources/ButtonActionMap.hpp:22]: Missing space before }

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Some typo

File: Client/ClientRoom.cpp (hunk 1 above)
Comment (osvegn) [reply Client/ClientRoom.cpp:148]: size allocated: sizeof(char) * 10
size used: (sizeof(char) * 5) * 2

File: libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateButton.hpp (hunk 3 above)
Comment (osvegn) [reply libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateButton.hpp:28]: I don't see any tabulation problem, can you explain ?

Comment (LucasTesnier) [reply libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateButton.hpp:28]: Strange, previously i was seeing a useless tab, but it's surely my Phone, i resolve the conversation

**Review by LucasTesnier — APPROVED** [review]

**Review by 31Nathan — APPROVED** [review]
LGTM

**Pull request merged**